### PR TITLE
[Frontend] Some console errors fixes on settings page

### PR DESCRIPTION
**Pull request by lndrtrbn**
### Proposed changes

* Diverse small fixes to clear a bit errors in dev console

### Related issues

* No related issues, just QoL for developers

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/components/ColorPickerField.jsx [hunk 1]
```diff
@@ -62,13 +62,15 @@ const ColorPickerField = (props) => {
...
   };
 
+  const { value, ...otherProps } = fieldToTextField(props);
+
   return (
     <>
       <MuiTextField
-        {...fieldToTextField(props)}
+        {...otherProps}
+        value={value ?? ''}
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/components/ColorPickerField.jsx:71]: Reason: `value` can be null. But an input should not be initialized with null or undefined value, instead use an empty string. By so, React can properly track changes with the input.

File: opencti-platform/opencti-front/src/components/MarkdownField.jsx [hunk 2]
```diff
@@ -63,7 +63,7 @@ const MarkdownField = (props) => {
         {label}
       </InputLabel>
       <ReactMde
-        value={field.value}
+        value={field.value ?? ''}
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/components/MarkdownField.jsx:66]: Reason: `value` can be null. But an input should not be initialized with null or undefined value, instead use an empty string. By so, React can properly track changes with the input.

File: opencti-platform/opencti-front/src/components/SelectField.jsx [hunk 3]
```diff
@@ -79,7 +81,8 @@ const SelectField = (props) => {
         {props.label}
       </InputLabel>
       <MuiSelect
-        {...fieldToSelect(props)}
+        {...otherProps}
+        value={value ?? ''}
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/components/SelectField.jsx:84]: Reason: `value` can be null. But an input should not be initialized with null or undefined value, instead use an empty string. By so, React can properly track changes with the input.

File: opencti-platform/opencti-front/src/components/TextField.jsx [hunk 4]
```diff
@@ -40,9 +40,12 @@ const TextField = (props) => {
     [onSubmit, setTouched, name],
   );
   const [, meta] = useField(name);
+  const { value, ...otherProps } = fieldToTextField(props);
+
   return (
     <MuiTextField
-      {...fieldToTextField(props)}
+      {...otherProps}
+      value={value ?? ''}
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/components/TextField.jsx:48]: Reason: `value` can be null. But an input should not be initialized with null or undefined value, instead use an empty string. By so, React can properly track changes with the input.

File: opencti-platform/opencti-front/src/private/components/settings/Settings.jsx [hunk 5]
```diff
@@ -484,7 +484,7 @@ const Settings = () => {
                                 <ItemBoolean
                                   variant="inList"
                                   neutralLabel={
-                                    settings.platform_cluster.instances_number
+                                    `${settings.platform_cluster.instances_number}`
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/settings/Settings.jsx:487]: Reason: `neutralLabel` accepts a string and not a number

File: opencti-platform/opencti-front/src/private/components/settings/roles/CapabilitiesList.tsx [hunk 6]
```diff
@@ -59,7 +59,7 @@ const CapabilitiesList: FunctionComponent<CapabilitiesListProps> = ({
             );
           }
         }
-        return <div key="none" />;
+        return <div key={i} />;
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/settings/roles/CapabilitiesList.tsx:62]: Reason: always set a unique key to elements inside a loop. By so, React can correctly track each individual element.

File: opencti-platform/opencti-front/src/private/components/settings/sub_types/SubTypesLine.tsx [hunk 7]
```diff
@@ -197,39 +197,34 @@ export const SubTypeLineDummy = ({
...
+                  animation="wave"
+                  variant="rectangular"
+                  width="90%"
+                  height={20}
+                />
+              </div>
+            ))}
+          </div>
+        }
+      />
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/settings/sub_types/SubTypesLine.tsx:227]: Just removed the imbricated `ListItem` as it was not required.

Reason: `li` inside `li` was generated.

File: opencti-platform/opencti-front/src/private/components/settings/roles/CapabilitiesList.tsx (hunk 6 above)
Comment (labo-flg) [reply opencti-platform/opencti-front/src/private/components/settings/roles/CapabilitiesList.tsx:62]: could you find something based on the data ? edge.node.id maybe ?

Using index is a last-resort strategy that can lead to issues (see https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/)

Comment (lndrtrbn) [reply opencti-platform/opencti-front/src/private/components/settings/roles/CapabilitiesList.tsx:62]: Well, this return is called when the data inside the map is undefined, so I can't really use any stuff linked to the data

Comment (labo-flg) [reply opencti-platform/opencti-front/src/private/components/settings/roles/CapabilitiesList.tsx:62]: ok then

**Review by labo-flg — APPROVED** [review]

**Pull request merged**